src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -107,7 +140,22 @@ public String getScript() {
         if (script.isEmpty()) {
             listener.getLogger().println("Warning: was asked to run an empty script");
         }
-        OsType os = ws.act(new getOsType());
+
+        final Jenkins jenkins = Jenkins.get();
+
+        Manifest manifest;
+        try (InputStream manifestStream = DurableTask.class.getResourceAsStream("/META-INF/MANIFEST.MF")) {
+            manifest = new Manifest(manifestStream);
+        }
+
+        Node wsNode = jenkins.getNode(FilePathUtils.getNodeName(ws));
+        FilePath nodeRoot = wsNode.getRootPath();
+
+        if (!Main.isUnitTest) {
+            PLUGIN_VERSION = manifest.getMainAttributes().getValue("Plugin-Version");
+        }
+        AgentInfo agentInfo = nodeRoot.act(new GetAgentInfo(nodeRoot, PLUGIN_VERSION));
```

Comment:
I don't think it makes sense to pass a `FilePath` to a `MasterToSlaveCallable` as an argument. You are already using `FilePath.act`, so if you change `GetAgentInfo` from a `MasterToSlaveCallable<AgentInfo,RuntimeException>` to a `FileCallable<AgentInfo>`, then instead of overriding `call()` you would override `invoke(File nodeRoot, VirtualChannel unused)`, and `invoke` allows you to throw `IOException` and `InterruptedException` so you don't have to handle that directly anymore.

Reply:
ahh, i see. Yeah,  that part bothered me using the filepath inside it. I just told myself that if i use filepath.act and it lives in the node, it would kind of be okay.

Reply:
Done in 5f7d04e.